modules/header-chain/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Defines traits which represent a common interface for Substrate pallets which want to
+//! incorporate bridge functinality.
+
+use frame_support::dispatch::DispatchResult;
+
+/// A base trait for tracking finalized headers on foreign chains in a Substrate pallet. Should be
+/// used as a primitive for more complicated header tracking logic.
+pub trait FinalityHeaderChain<AccountId> {
```

Comment:
I think the crate should rather be in `primitives` not in `modules`.

Reply:
I thought about this and was a little conflicted. Ultimately I decided not to put in `primitives` because it's not used by the relay at all - it's only used by pallets.

Reply:
I like the split we have in substrate, where `primitives` are things that are cheap to import since they mostly define traits + data types and are often shared. That's my reasoning to put it there :)